### PR TITLE
Much faster implementation

### PR DESCRIPTION
**Pull request by samuelgozi**
I know this is a lazy PR since I didn't check other files to make sure I'm making the correct changes, so I'm sorry about that.
This implementation is much faster.
The benchmark checks the time to generate one million nanoids.
```
$ node ./main.js
Old 1365ms
New 948ms

$ bun ./main.js
Old 462ms
New 138ms
```

I hope you find it useful 🙏

### CODE REVIEW

**Comment by ai** [comment]
1. It is the file generated from `index.browser.js`. You need to fix it first.
2. But browser's file should be tested in browsers. For Node/Bun we have another file `index.js` which has different optimizations.

**Comment by samuelgozi** [comment]
Tested on latest version:

Chrome
```
Old 1634ms
New 1245ms
```

Safari
```
Old 1208ms
New 760ms
```

Firefox
```
Old 1207
New 742
```

Browser support doesn't change much.
I will update the source file.

**Comment by samuelgozi** [comment]
I now updated the right place.

File: nanoid.js [hunk 1]
```diff
@@ -1 +1 @@
-export let nanoid=(t=21)=>crypto.getRandomValues(new Uint8Array(t)).reduce(((t,e)=>t+=(e&=63)<36?e.toString(36):e<62?(e-26).toString(36).toUpperCase():e>62?"-":"_"),"");
```
Comment (ai) [inline nanoid.js:1]: Could you revert changes? I will update it by script.

File: index.browser.js [hunk 2]
```diff
@@ -47,25 +47,15 @@ export let customRandom = (alphabet, defaultSize, getRandom) => {
...
-      id += '_'
-    }
-    return id
-  }, '')
-
+export let nanoid = (len = 21) => {
+	let id = "";
+  // This will be hoisted by the JIT engine.
+  // I keep it here to make the code compact and easy to copy.
+	const chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";
```
Comment (ai) [inline index.browser.js:54]: Should we move it outside of `nanoid()` function?

File: index.browser.js [hunk 3]
```diff
@@ -47,25 +47,15 @@ export let customRandom = (alphabet, defaultSize, getRandom) => {
...
-    } else if (byte > 62) {
-      id += '-'
-    } else {
-      id += '_'
-    }
-    return id
-  }, '')
-
+export let nanoid = (len = 21) => {
+	let id = "";
```
Comment (ai) [inline index.browser.js:51]: 1. Let’s fix code style issues from ESLint.
2. And we will need to update `package→size-limit` to see how much we increase the JS bundle size (for browser’s version we are optimizing both size and performance)

**Comment by samuelgozi** [comment]
Ok i will fix everything you mentioned.
In regards to the use of the string inside the function, I don't think we should hoist it manually since I didn't see any effect on performance, but I do think that this way it looks nicer.

Maybe I should remove the comment instead since it not important

**Comment by ai** [comment]
OK, let's keep string as is. I will do some optimizations later for smaller bundle size.

**Comment by samuelgozi** [comment]
Done. Package size limit has exceeded by 34 B

**Comment by ai** [comment]
> Package size limit has exceeded by 34 B

Try to import alphabet from https://github.com/ai/nanoid/blob/main/url-alphabet/index.js

We have a fun hack there, we put the same letters not in alhabetical order, but in order for better gzip.

And please update `packge→size-limit` for new size (so we will track size changes by commit).

**Comment by samuelgozi** [comment]
`Package size limit has exceeded by 15 B`
TIL, very cool!

Made the changes

**Comment by ai** [comment]
I also reduced 8 bytes by reusing `size` variable 4db3464

**Comment by samuelgozi** [comment]
Using the while loop is actually a little bit faster.
Very nice idea!

**Pull request merged**